test/unit/modules/util/next-tick.spec.js

```diff
@@ -0,0 +1,25 @@
+import { nextTick } from 'core/util/env'
+
+describe('nextTick', () => {
+  it('accepts a callback', done => {
+    nextTick(done)
+  })
+
+  it('returns undefined when passed a callback', () => {
+    expect(typeof nextTick(() => {})).toBe('undefined')
+  })
+
+  if (typeof Promise !== 'undefined') {
+    it('returns a Promise when provided no callback', done => {
+      nextTick().then(done)
+    })
+
+    it('returns a Promise with a context argument when provided a falsy callback and an object', done => {
+      const obj = {}
+      nextTick(undefined, obj).then(ctx => {
```

Comment:
why ctx is passed as argument, I wonder. Because `nextTick` does not modify obj, the caller of `nextTick` can always access `ctx`. If we are using `nextTick` as a `Promise` fallback in which `ctx` is wrapped and consumer does not own handle to `ctx`, it is possibly useless because we can use native `Promise`. If promise does not exist, nextTick returns `undefined` which does add much value.